runtime/vm/gphandle.c

```diff
@@ -893,7 +893,7 @@ generateSystemDump(struct J9PortLibrary* portLibrary, void* gpInfo)
 	PORT_ACCESS_FROM_PORT(portLibrary);
 
 	getEnvForDumpReturnCode = j9sysinfo_get_env(J9DO_NOT_WRITE_DUMP, NULL, 0);
-	if (getEnvForDumpReturnCode != 0) {
+	if (getEnvForDumpReturnCode == -1) {
```

Comment:
While you're here, please tidy up:
* initialize local variables
* put the constant on the left of `==`
* declare variables in the scope in which they're used
```
	PORT_ACCESS_FROM_PORT(portLibrary);
	IDATA getEnvForDumpReturnCode = j9sysinfo_get_env(J9DO_NOT_WRITE_DUMP, NULL, 0);

	if (-1 == getEnvForDumpReturnCode) {
		char dumpName[EsMaxPath];
		UDATA dumpCreateReturnCode = 0;

		/* failed to find the env var, so write the dump */
		*dumpName = '\0';
#ifdef LINUX
		/* Ensure that userdata argument is NULL on Linux. */
		dumpCreateReturnCode = j9dump_create(dumpName, NULL, NULL);
#else
		dumpCreateReturnCode = j9dump_create(dumpName, NULL, gpInfo);
#endif

		if (0 == dumpCreateReturnCode) {
```

Reply:
Updated the code as suggested.